### PR TITLE
Store Selected Functions as Values

### PR DESCRIPTION
**Pull request by florian-kuebler**
This is the first part of a series of moving away from representing
functions by absolute addresses in order to make them independent
of the loaded process.

Prior to this, DataManager stored the selected functions as
as set of their absolute addresses.
This changes this, and stores them by value.
As protos have no hash/equals implementation, FunctionInfoSet
was added.

Test: Compile
Bug: http://b/169309553

### CODE REVIEW

**Review by dimitry- — COMMENTED** [review]
are there any advantages in storing selected functions as values?

**Comment by florian-kuebler** [comment]
> are there any advantages in storing selected functions as values?

The idea is the following: We currently store them by absolute address. The long-term idea of that series of changes, is to make function/modules independent from process. So we also want to restrict using absolute addresses (there are some places, where we can't and shouldn't get rid of them, but if it makes sense, we should).
I was thinking of storing the functions as pointers (this is how this was done in the past at some point in time),
but we now have different locations, where the function might be owned, so we can't rely on pointers.

File: OrbitClientData/FunctionInfoSetTest.cpp [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  left.set_loaded_module_path("/path/to/module");
+  left.set_module_base_address(42);
+  left.set_address(12);
+  left.set_load_bias(4);
+  left.set_size(16);
+  left.set_file("file.cpp");
+  left.set_line(13);
+
+  FunctionInfo right;
+  right.CopyFrom(left);
```
Comment (dimitry-) [inline OrbitClientData/FunctionInfoSetTest.cpp:25]: maybe just copy and paste in this particular case lines 13-22 for right?

**Review by dimitry- — APPROVED** [review]
otherwise lgtm

File: OrbitClientData/FunctionInfoSetTest.cpp (hunk 1 above)
Comment (florian-kuebler) [reply OrbitClientData/FunctionInfoSetTest.cpp]: sure. Done

**Comment by florian-kuebler** [comment]
Thanks for the review

**Pull request merged**